File: csharp/secret-handshake/SecretHandshake.cs
using System.Collections.Generic;
using System.Linq;

public static class SecretHandshake
{
    public static string[] Commands(int commandValue)
    {
        var wink = (commandValue & 0b0001) != 0;
        var doubleBlink = (commandValue & 0b0010) != 0;
        var closeYourEyes = (commandValue & 0b0100) != 0;
        var jump = (commandValue & 0b1000) != 0;
        var reverse = (commandValue & 0b10000) != 0;

        var result = new List<string>(4);

        if (wink)
        {
            result.Add("wink");
        }

        if (doubleBlink)
        {
            result.Add("double blink");
        }

        if (closeYourEyes)
        {
            result.Add("close your eyes");
        }

        if (jump)
        {
            result.Add("jump");
        }

        if (reverse)
        {
            result.Reverse();
        }

        return result.ToArray();
    }
}

File: go/secret-handshake/secret_handshake.go
package secret

import (
	"slices"
)

func Handshake(code uint) []string {
	result := []string{}

	if (code & 0x01) != 0 {
		result = append(result, "wink")
	}

	if (code & 0x02) != 0 {
		result = append(result, "double blink")
	}

	if (code & 0x04) != 0 {
		result = append(result, "close your eyes")
	}

	if (code & 0x08) != 0 {
		result = append(result, "jump")
	}

	if (code & 0x10) != 0 {
		slices.Reverse(result)
	}

	return result
}
